### PR TITLE
[Job Launcher] feat: get rid of sdk's storage module

### LINKED ISSUE #2718: [Job Launcher] Get rid of StorageModule usage
**Description**
Remove StorageModule usage and replace it's usages.

**Motivation**
StorageModule from our SDK is deprecated and is going to be deleted soon.

**Implementation details** 
TBD by developer

### PR DESCRIPTION
**Pull request by dnechay**
## Issue tracking
Closes #2718 

## Context behind the change
Storage module in SDK is going to be deprecated, so we get rid of it here. At the same time existing storage utils are not enough for working with binaries (e.g. to download & decrypt job results as `.zip`), so we prepare the ground in this PR.

## How has this been tested?
- [x] Existing unit tests pass
- [x] Test all methods on local using local Minio setup and testing endpoints
- [x] Manual tests that `downloadFileFromUrl` throws expected errors

## Release plan
1. Drop commit with "test" functionality & force push (`do-not-merge` label as reminder)
2. Merge & deploy

## Potential risks; What to monitor; Rollback plan
Some implicit bugs related to cleanup of `download` usage (e.g. double `JSON.parse` code paths)

### CODE REVIEW

File: packages/apps/job-launcher/server/docker-compose.yml [hunk 1]
```diff
@@ -38,7 +38,7 @@ services:
         condition: service_healthy
     entrypoint: >
       /bin/sh -c "
-      /usr/bin/mc config host add myminio http://minio:9000  access-key secret-key;
+      /usr/bin/mc config host add myminio http://minio:9000 minioadmin minioadmin;
```
Comment (dnechay) [inline packages/apps/job-launcher/server/docker-compose.yml:41]: These are default creds and below cmd creates bucket, so now minio works properly out of the box

File: packages/apps/job-launcher/server/src/common/interceptors/snake-case.ts [hunk 2]
```diff
@@ -21,8 +22,13 @@ export class SnakeCaseInterceptor implements NestInterceptor {
       request.query = CaseConverter.transformToCamelCase(request.query);
     }
 
-    return next
-      .handle()
-      .pipe(map((data) => CaseConverter.transformToSnakeCase(data)));
+    return next.handle().pipe(
+      map((data) => {
+        if (data instanceof StreamableFile) {
```
Comment (dnechay) [inline packages/apps/job-launcher/server/src/common/interceptors/snake-case.ts:27]: W/o this files can't be downloaded: `StreamableFile` gets modified to snake case and just returned as `json`

File: packages/apps/job-launcher/server/src/modules/job/job.service.ts [hunk 3]
```diff
@@ -1326,10 +1306,11 @@ export class JobService {
     }
 
     if (jobEntity.requestType === JobRequestType.FORTUNE) {
-      const data = await this.storageService.download(finalResultUrl);
-      const result = typeof data === 'string' ? JSON.parse(data) : data;
+      const data = (await this.storageService.downloadJsonLikeData(
+        finalResultUrl,
+      )) as any;
 
-      if (!result) {
```
Comment (dnechay) [inline packages/apps/job-launcher/server/src/modules/job/job.service.ts:1332]: When file is not found, we get `[]` (both before and after refactoring), so seems like we never get to this condition.

Should we define the type of `data` here as array of fortune results?

File: packages/apps/job-launcher/server/src/modules/job/job.service.ts [hunk 4]
```diff
@@ -1496,25 +1479,6 @@ export class JobService {
       );
     }
 
-    let manifest;
-    if (typeof manifestData === 'string' && isPGPMessage(manifestData)) {
```
Comment (dnechay) [inline packages/apps/job-launcher/server/src/modules/job/job.service.ts:1500]: Same as https://github.com/humanprotocol/human-protocol/pull/2756/files#r1829448115

File: packages/apps/job-launcher/server/src/modules/storage/storage.service.ts [hunk 5]
```diff
@@ -28,57 +31,116 @@
...
+
+  public static async downloadFileFromUrl(url: string): Promise<Buffer> {
+    if (!this.isValidUrl(url)) {
+      throw new InvalidFileUrl(url);
+    }
+
+    try {
+      const { data } = await axios.get(url, {
+        responseType: 'arraybuffer',
+      });
```
[the remark this answers is not in the document]
Comment (dnechay) [reply packages/apps/job-launcher/server/src/modules/storage/storage.service.ts:52]: We will drop commit with test controller before merge, so this alert should be gone after that

**Review by Dzeranov — COMMENTED** [review]
lgtm
@portuu3 please take a look

File: packages/apps/job-launcher/server/docker-compose.yml (hunk 1 above)
Comment (portuu3) [reply packages/apps/job-launcher/server/docker-compose.yml:41]: wasn't this working before?

Comment (dnechay) [reply packages/apps/job-launcher/server/docker-compose.yml:41]: @portuu3 by default - no. In order to make it working you have to either:
- start minio, go to console, create a service account with this access/secret key
- or have a config file for minio to start container with access/secret key defined

I think it's easier to use default credentials from minio itself, esp taking into account it's only for local development

File: packages/apps/job-launcher/server/src/modules/job/job.service.ts (hunk 3 above)
Comment (portuu3) [reply packages/apps/job-launcher/server/src/modules/job/job.service.ts:1332]: > When file is not found, we get [] (both before and after refactoring), so seems like we never get to this condition.

Maybe we can check the array's lenght

> Should we define the type of data here as array of fortune results? 

lgtm

Comment (dnechay) [reply packages/apps/job-launcher/server/src/modules/job/job.service.ts:1332]: @portuu3 
Done in https://github.com/humanprotocol/human-protocol/pull/2756/commits/a8bdcf9171a267c573c35e43ca46546164df63d1

**Review by portuu3 — DISMISSED** [review]

File: packages/apps/job-launcher/server/src/modules/storage/storage.controller.ts
[the remark this answers is not in the document]
Comment (dnechay) [reply packages/apps/job-launcher/server/src/modules/storage/storage.controller.ts:1]: Yup, it's not there anymore

**Review by portuu3 — APPROVED** [review]

**Pull request merged**